json.unmarshal errors on arrays when assertions are disabled

## Context

* Operating System & Odin Version: Windows 11 & Odin dev-2022-07:7c990b38 (master branch checkout)
* Please paste `odin report` output:

```
        Odin: dev-2022-07:7c990b38
        OS:   Windows 11 Professional (version: 21H2), build 22000.795
        CPU:  Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz
        RAM:  16238 MiB
```

I've tested this also on macOS, with the same result.

## Expected Behavior

Marshaling and unmarshaling a struct should work without error, even with assertions disabled.

## Current Behavior

When assertions are disabled via the cli flag `-disable-assert` (either using `odin run` or `odin build`), unmarshaling a JSON value with arrays in it errors.

## Failure Information (for bugs)

### Steps to Reproduce

Here is the code necessary to reproduce this:

```odin
/*
Works when assertions are enabled:

    > odin run marshal_problems.odin -file
	[INFO ] --- [2022-07-19 12:45:04] [marshal_problems.odin:45:main()]  All good

Breaks when assertions are disabled:

    > odin run marshal_problems.odin -file -disable-assert
	[ERROR] --- [2022-07-19 12:57:50] [marshal_problems.odin:40:main()]  Unsupported_Type_Error{id = [2]int, token = Token{pos = Pos{offset = 11, line = 1, column = 11}, kind = "Colon", text = ":"}}
*/
package main

import "core:encoding/json"
import "core:log"
import "core:os"

Settings :: struct {
	position: [2]int,
	size: [2]int,
}

main :: proc() {
	context.logger = log.create_console_logger(lowest = .Info)

	settings := Settings {
		[2]int{10, 50},
		[2]int{150, 970},
	}

	blob, err := json.marshal(settings)
	if err != nil {
		log.error(err)
		os.exit(1)
	}

	new_settings : Settings = ---
	err2 := json.unmarshal(blob, &new_settings)
	if err2 != nil {
		log.error(err2)
		os.exit(1)
	}

	log.info("All good")
}
```

--- a/core/encoding/json/unmarshal.odin
+++ b/core/encoding/json/unmarshal.odin
@@ -325,7 +325,7 @@ unmarshal_object :: proc(p: ^Parser, v: any, end_token: Token_Kind) -> (err: Unm
 	UNSUPPORTED_TYPE := Unsupported_Type_Error{v.id, p.curr_token}
 	
 	if end_token == .Close_Brace {
-		assert(expect_token(p, .Open_Brace) == nil)
+		unmarshal_expect_token(p, .Open_Brace)
 	}
 
 	v := v
@@ -473,7 +473,7 @@ unmarshal_object :: proc(p: ^Parser, v: any, end_token: Token_Kind) -> (err: Unm
 	}
 	
 	if end_token == .Close_Brace {
-		assert(expect_token(p, .Close_Brace) == nil)
+		unmarshal_expect_token(p, .Close_Brace)
 	}
 	return
 }

